impexp-client/src/main/java/org/citydb/cli/options/vis/DisplayOption.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citydb.cli.options.vis;
+
+import org.citydb.config.project.kmlExporter.DisplayForm;
+import org.citydb.config.project.kmlExporter.KmlExportConfig;
+import org.citydb.plugin.cli.CliOption;
+import picocli.CommandLine;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DisplayOption implements CliOption {
+    @CommandLine.Option(names = {"-D", "--display-mode"}, split = ",", paramLabel = "<mode>", required = true,
+            description = "Display mode: ${COMPLETION-CANDIDATES}.")
+    private Set<Mode> modes;
+
+    @CommandLine.Option(names = {"-l", "--lod"}, paramLabel = "<0..4 | halod>", required = true,
+            description = "LoD to export from.")
+    private Lod lod;
+
+    @CommandLine.Option(names = {"-v", "--visible-from"}, split = ",", paramLabel = "<mode=value>",
+            description = "Visibility for each display mode (default: 0).")
+    private Map<Mode, Integer> visibleFrom;
+
+    @CommandLine.Option(names = {"-a", "--appearance-theme"}, paramLabel = "<theme>",
+            description = "Appearance theme to use for COLLADA/glTF exports. Use 'none' for the null theme.")
+    private String theme;
+
+    public Set<Mode> getModes() {
+        return modes;
+    }
+
+    public int getLod() {
+        return lod.ordinal();
+    }
+
+    public String getAppearanceTheme() {
+        if (theme == null) {
+            return KmlExportConfig.THEME_NONE;
+        }
+
+        return "none".equalsIgnoreCase(theme) ? KmlExportConfig.THEME_NULL : theme;
+    }
+
+    public void toDisplayForms(List<DisplayForm> displayForms) {
+        Map<Integer, DisplayForm> candidates = displayForms.stream()
+                .collect(Collectors.toMap(DisplayForm::getForm, d -> d));
+
+        int visibleTo = -1;
+        for (Mode mode : Mode.values()) {
+            DisplayForm displayForm = candidates.computeIfAbsent(mode.type, v -> new DisplayForm(mode.type, -1, -1));
+            displayForm.setActive(modes.contains(mode) && lod.ordinal() >= mode.minimumLod);
+            if (displayForm.isActive()) {
+                int visibleFrom = this.visibleFrom != null ? this.visibleFrom.getOrDefault(mode, 0) : 0;
+                displayForm.setVisibleFrom(visibleFrom);
+                displayForm.setVisibleUpTo(visibleTo);
+                visibleTo = visibleFrom;
+            }
+        }
+
+        candidates.values().forEach(displayForm -> {
+            if (!displayForms.contains(displayForm)) {
+                displayForms.add(displayForm);
+            }
+        });
+    }
```

Comment:
The current implementation requires for example the following syntax:

```
--display-mode=collada,geometry --visible-from=extruded=120,geometry=120,collada=120
```

The modes given in `--display-mode` are prioritized over those given in `--visible-from`, which means in this case `extruded=120` will be ignored. Would it be better/faster to exclude `--display-form` and use the modes given in `--visible-from` instead, so that users do not need to enter the modes twice? In this case perhaps `--visible-from` would be renamed to something like `mode-visibility`, etc.

Otherwise I'd suggest adding the following line (or similar) to the description of `--visble-from` in line 51:

```
To define modes, use --display-mode.
```

so that it is clear to the users that `--visible-from` does not automatically add modes to the display forms, only those given in `--display-mode` are accepted.

Reply:
Yes, currently you must define the export format via `--display-mode`. Since the visibility setting is only optional (default: 0), I moved it to its own parameter `--visible-from`, which must therefore refer to the formats defined in `--display-mode`. You cannot add further display modes through `--visible-from`.

Both options could be merged into one: `display-mode=MODE[=VISIBLE_FROM],MODE[=VISIBLE_FROM]...`

For example:
```
--display-mode=collada // just export collada with visible from = 0
--display-mode=collada=50,footprint=100
```

The `picocli` library I m using does not support such a value list out-of-the-box, so it requires writing a specific option converter. I tried to avoid this with the current implementation.

Reply:
> Both options could be merged into one: ``display-mode=MODE[=VISIBLE_FROM],MODE[=VISIBLE_FROM]...``

This would be ideal for the next release(s) then.

Reply:
I'd like to make this change **before** the next release, but outside this PR. The reason is that I reworked the config settings for KML/COLLADA/glTF exports in https://github.com/3dcitydb/importer-exporter/pull/149 because there was a lot of duplicate code, especially concerning the display forms. So implementing this proposal will be a lot easier after we have merged https://github.com/3dcitydb/importer-exporter/pull/149.

Reply:
Agreed.